lib/src/main/kotlin/org/didcommx/didcomm/message/Message.kt

```diff
@@ -27,10 +28,50 @@ data class Message(
     val pthid: String?,
     val customHeaders: Map<String, Any?>,
 ) {
+    var forwardNext: String? = null
```

Comment:
Not sure it's good idea to mix the Message and ForwardMessage. 
1) The Forward protocol may change, and we will have to change the core message
2) Message is part of a public API. Forward is rather advanced or internal one. So, every time a user gets a message, he will see these weird forwardNext and forwardedMsg fields.
3) Having mutable fields when other fields are immutable looks suspicious. Also, for a person who expects a Forward message, the fields are nullable, so additional null checks will be required.